src/matches.rs

```diff
@@ -143,3 +145,20 @@ fn has_only_ref_pats(arms: &[Arm]) -> bool {
     // look for Some(v) where there's at least one true element
     mapped.map_or(false, |v| v.iter().any(|el| *el))
 }
+
+fn match_template(source: MatchSource, op: &str, expr: &str) -> String {
+    match source {
+        MatchSource::Normal => {
+            format!("match {}{} {{ ...", op, expr)
+        }
+        MatchSource::IfLetDesugar { .. } => {
+            format!("if let ... = {}{} {{", op, expr)
+        }
+        MatchSource::WhileLetDesugar => {
+            format!("while let ... = {}{} {{", op, expr)
+        }
+        MatchSource::ForLoopDesugar => {
+            panic!("for loop desugared to match with &-patterns!")
```

Comment:
I'd avoid for loops since `for x in &foo` is often done which may cause problems.

Either that or add enough tests to ensure that this doesn't have false positives :smile: